packages/main/src/DateRangePicker.js

```diff
@@ -92,21 +92,35 @@ class DateRangePicker extends DatePicker {
 	constructor() {
 		super();
 		this.isFirstDatePick = true;
+		this._initialRendering = true;
+		this._oneTimeStampSelected = false; // Used to determine whether the first & last date is the same
 	}
 
 	async onAfterRendering() {
-		this.responsivePopover = await this._respPopover();
-		const calendar = this.responsivePopover.querySelector(`#${this._id}-calendar`);
-		const dayPicker = calendar.shadowRoot.querySelector(`#${calendar._id}-daypicker`);
-		dayPicker.addEventListener("item-mouseover", this._itemMouseoverHandler);
-		dayPicker.addEventListener("daypickerrendered", this._keyboardNavigationHandler);
+		this.daypicker = this.getDaypicker();
+		this._cleanHoveredAttributeFromVisibleItems(this.daypicker);
+		this._initialRendering = false;
+	}
+
+	async onEnterDOM() {
+		this.daypicker = await this.getDaypicker();
```

Comment:
Is there benefit of assignment the following to "this":
this.daypicker = await this.getDaypicker();

I could not find to be used somewhere else. I think you need a local variable (same is valid for onAfterRendering and onExitDOM).

```js
const dayPicker = await this.getDaypicker();
dayPicker.addEventListener
dayPicker.addEventListener
```

...perhaps getDaypicker can become getDayPicker, but not necessarily.

Reply:
The benefit is not the assignment, but rather the fact that we only have the code for getting the daypicker at one place, which is easier to debug & support. If you insist, I can use the code without a method and put it in 3 places.